src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.Helpers.cs

```diff
@@ -12,26 +13,31 @@ namespace Microsoft.NET.Sdk.Razor.SourceGenerators
 {
     public partial class RazorSourceGenerator
     {
-        internal static string GetIdentifierFromPath(string filePath)
+        internal static string GetIdentifierFromPath(ReadOnlySpan<char> filePath)
         {
-            var builder = new StringBuilder(filePath.Length);
+            using var _ = ArrayPool<char>.Shared.GetPooledArraySpan(filePath.Length + 5, out var destination);
```

Comment:
>GetPooledArraySpan [](http://example.com/codeflow?start=49&length=18)

Why not just use StringBuilderPool?

Reply:
It would be a bit nicer from my perspective, but I don't have a good perf justification as both ways achieve the allocation reduction